openedx_ledger/test_utils/factories.py

```diff
@@ -37,6 +44,17 @@ class Meta:
     lms_user_id = factory.Faker("random_int", min=1, max=1000)
 
 
+class ExternalTransactionReferenceFactory(factory.django.DjangoModelFactory):
+    """
+    Test factory for the `ExternalTransactionReferenceFactory` model.
+    """
+    class Meta:
+        model = ExternalTransactionReference
+
+    external_reference_id = factory.Faker("lexify", text="????-????-????-????")
+    transaction = factory.SubFactory(TransactionFactory)
```

Comment:
should creating a reference really trigger the creation of a transaction?  What if instead you used iterator:
```suggestion
    transaction = factory.Iterator(Transaction.objects.all())
```

Reply:
When would an external reference exist without an associated transaction? My understanding was that there **should** be associated records with external references and/or If you wanted to, could you not supply `None` to the method if that specific situation called for it? Personally I feel rather indifferent between the two approaches ¯\_(ツ)_/¯ so willing to make the change if you feel strongly

Reply:
Fundamentally, I agree with what you're implying: that an external reference should probably not exist without an associated transaction.  Both of these factory defaults for the transaction FK fulfills that need, but the one I propose just uses an existing transaction rather than creating a new one.

BTW, in your subsidy PR, the unit test you added first creates an ExternalTransactionReference (without supplying a transaction to the factory), then creates a Transaction, then does surgery to link the reference to the transaction.  That's all fine and works, but that has the side effect of creating an orphaned transaction that never gets used.  My suggested approach would have forced you to refactor the unit test code to create the transaction first, followed by the external reference, avoiding creating any orphaned transaction.

Reply:
Makes sense to me!